sw/device/silicon_creator/lib/shutdown.c

```diff
@@ -0,0 +1,277 @@
+#include "sw/device/silicon_creator/lib/shutdown.h"
+
+#include <assert.h>
+
+#include "sw/device/lib/base/bitfield.h"
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/stdasm.h"
+#include "sw/device/lib/runtime/print.h"
+#include "sw/device/silicon_creator/lib/base/abs_mmio.h"
+#include "sw/device/silicon_creator/lib/drivers/alert.h"
+#include "sw/device/silicon_creator/lib/drivers/lifecycle.h"
+
+#include "alert_handler_regs.h"
+#include "flash_ctrl_regs.h"
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "keymgr_regs.h"
+#include "otp_ctrl_regs.h"
+#include "sram_ctrl_regs.h"
+
+
+static_assert(
+  ALERT_HANDLER_ALERT_CLASS_MULTIREG_COUNT <= OTP_CTRL_PARAM_ROM_ALERT_CLASSIFICATION_SIZE / 4,
+  "More alerts than alert classification OTP words!");
+
+
+#ifdef MOCK_ABS_MMIO
+// If we're building as a unit test, rename the shutdown functions so they
```

Comment:
I think I understand what you're doing here, but it might be helpful to elaborate further...?

Reply:
Done.